### DISCUSSION TITLE
Version 0.6.8: Launch config builder improvement, CUDA 12.x kernel-library support and launch attributes

### DISCUSSION THREAD

**Discussion by eyalroz** [eyalroz/cuda-api-wrappers#584] [Announcements]
Changes since v0.6.7:

## Build process & build configuration changes

* #583 Exported targets now make sure apps link against some system libraries CUDA depends on (to circumvent CMake bug [25665](https://gitlab.kitware.com/cmake/cmake/-/issues/25665)).
* #567 The runtime-and-driver target now has `driver-and-runtime` as an alias

## Launch configuration & launch config builder changes

* #564 Can now launch kernels with the full range of CUDA 12.x launch attributes, including remote memory sync domain, programmatic launch dependence and programmatic completion events (see descriptions in the [CUDA Driver API documentation](https://docs.nvidia.com/cuda/cuda-driver-api/group__CUDA__TYPES.html#group__CUDA__TYPES_1g6f6565b334be6bb3134868e10bbdd331)).
* #577, #582 More extensive validation of launch configurations when building them with the launch config builder gadget.
* #581 More robust comparison operators for dimension structures
* #580 Launch config builder can now be told to the "use the maximum number of active blocks per multiprocessor".
* #579 User can now set a target device on a launch configuration target device without setting a contextualized kernel, in a launch config builder
* #578 Now using the launch config builder in more of the example programs
* #569 Took care of unused validation function which was triggering a warning with newer compilers

## CUDA libraries and in-library, non-associated kernel support

* #565 Now supporting "CUDA libraries" - files or blocks of data in memory containing compiled kernels, which are _not_ loaded immediately into modules within contexts; and contain device-and-context-independent compiled kernels. Both of these can now be represented and worked with.
* #576 A module no longer holds the link options it was created with; those are not essential to its use, and at times are impossible to (re)create when obtaining a module from a library (which also doesn't hold its link options ).

## Other changes

* #575 Moved `apriori_compiled_kernel_t` into the `kernel` namespace, yielding  `kernel::apriori_compiled_it`
* #570, #573 Removed some redundant inclusions and definitions
* #568 Fixed some breakage of `kernel_t::set_attribute()`
* #566 Can now properly get and set properties on kernels (raw functions and handles)

## Compatibility

* #572 Fixed broken CUDA 9.x compatibility

 --- 

* Want to help me with testing 0.7? Drop me a line... (it will have CUDA execution graph support)
* Have you tried this version and are satisfied with the changes? Thumb-up just below this line to let others know as well.

<hr /><em>This discussion was created from the release <a href='https://github.com/eyalroz/cuda-api-wrappers/releases/tag/v0.6.8b1'>Version 0.6.8: Launch config builder improvement, CUDA 12.x kernel-library support and launch attributes</a>.</em>

**Comment by crossterra-agent** [comment]
- **Status**: DONE
- **Role**: orchestrator
- **Step**:
- **Link**: https://github.com/CrossTerra/monorepo/pull/1923
- **Summary**: Created PR with developer instructions for size guide database tables (S01–S05).
